### PR TITLE
feat: [SPG-191]: Have account id in query for artifacts

### PR DESCRIPTION
**Pull request by abhinav-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38093)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38093)

**Review by inderschera — COMMENTED** [review]
> Reviewed 1 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38093)* status: 2 file(s) to review. 0 of 1 approvals obtained. 

___
*[`400-rest/src/test/resources/mongo/indexes.txt` line 75 at r4](https://reviewable.io/reviews/harness/harness-core/38093#-NDGt9nKD4E6cXO1MQLS:-NDGt9nKD4E6cXO1MQLT:b-jgsp5f) ([raw file](https://github.com/harness/harness-core/blob/9240ee0f403c52ba39cdfdcd478128b21333c6ea/400-rest/src/test/resources/mongo/indexes.txt#L75)):*
> ```help
> artifacts {"name": "accountId_1", "background": true} {"accountId": 1}
> artifacts {"name": "accountId_artifactStreamId_createdAt", "background": true} {"accountId": 1, "artifactStreamId": 1, "createdAt": -1}
> artifacts {"name": "accountId_createdAt_artifactStreamIds", "background": true} {"accountId": 1, "createdAt": -1, "artifactStreamId": 1}
> ```

why do we need this index? I already see accountId_artifactStreamId_createdAt index just above it.

File: 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java [hunk 1]
```diff
@@ -2843,7 +2844,9 @@ private LastDeployedArtifactInformation populateLastDeployedArtifactInfo(List<Ar
 
   private boolean isArtifactPresentInStream(Optional<Artifact> requiredArtifact, String serviceId, String appId) {
     if (requiredArtifact.isPresent()) {
-      List<Artifact> presentArtifacts = artifactService.listArtifactsForService(appId, serviceId, new PageRequest<>());
+      PageRequest pageRequest = new PageRequest<>();
+      pageRequest.addFilter(ArtifactKeys.accountId, EQ, requiredArtifact.get().getAccountId());
```
Comment (ershadalim) [inline 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java:2848]: This is will either require a new UT or modify existing one

**Review by Y0709 — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 2 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38093)* status: :shipit: complete! ✓

**Pull request merged**